salt/orch/remove-nodes.sls

```diff
@@ -0,0 +1,46 @@
+{%- set removed_nodes = pillar.get('caasp_removed_nodes', []) -%}
+
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+update_grains:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_grains
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+       - salt: update_grains
+
+{%- for node_id in removed_nodes %}
+
+{{ node_id }}-stop-services:
+  salt.state:
+    - tgt: {{ node_id }}
+    - sls:
+      - container-feeder.stop
+      - kube-apiserver.stop
+      - kube-controller-manager.stop
+      - kube-scheduler.stop
+      - kube-proxy.stop
+      - kubelet.stop
+      - docker.stop
+      - etcd.stop
```

Comment:
I also think that as part of the strategy we could disable the `salt-minion` service and poweroff or shutdown the machine after we have made sure the rest of the cluster is fine. So, something like:

- Mark nodes as "to be removed" from the UI (using `salt-api`, writing a grain on the machines)
- Call to the `orch.remove-nodes` orchestration from the UI (using `salt-api`)
- The orchestration:
    - Stops the basic kubernetes services and remove private information (preferably using `shred`)
    - Makes sure the rest of the nodes are working as expected after this services has been stopped (maybe running some highstates on the remaining nodes?)
    - Disables the `salt-minion` service (can't stop it because we'll still need it until we poweroff, or reboot)
    - Reboot or poweroff the machine

Reply:
I was considering `shred` as command for cleaning data, but since we are operating on /root disk, it will be hard to implement. Probably `wipe` command will be better, but I think that `rm -fr` is enough.